File: files/samples/Query_Content.py
from native import alert
from native import app
from native import json

def OnStart():
    lay = app.CreateLayout("linear", "VCenter,FillXY")

    btnPhone = app.CreateButton("Get Phone Number", 0.4, 0.1)
    btnPhone.SetOnTouch(btnPhone_OnTouch)
    lay.AddChild(btnPhone)

    btnContact = app.CreateButton("Get D Contacts", 0.4, 0.1)
    btnContact.SetOnTouch(btnContact_OnTouch)
    lay.AddChild(btnContact)

    app.AddLayout(lay)

def btnPhone_OnTouch():
    action = "android.intent.action.PICK"
    uri = "content://contacts"
    type = "vnd.android.cursor.dir/phone_v2"
    app.SendIntent(None, None, action, None, uri, type, None, "result", OnResult)

def OnResult(resultCode, result):
    rows = app.QueryContent(result.data, "data1, display_name")
    app.ShowPopup(rows[0].display_name + " " + rows[0].data1)

def btnContact_OnTouch():
    uri = "content://com.android.contacts/data"
    columns = "display_name, data1"
    select = "display_name LIKE ?"
    params = ["D%"]
    rows = app.QueryContent(uri, columns, select, params)
    app.Alert(str(rows))

File: files/samples/Query_Content.js

//Called when application is started.
function OnStart()
{
	//Create a layout with objects vertically centered.
	lay = app.CreateLayout( "linear", "VCenter,FillXY" )	
	
	//Create an contacts intent button.
	btnPhone = app.CreateButton( "Get Phone Number", 0.4, 0.1 )
	btnPhone.SetOnTouch( btnPhone_OnTouch )
	lay.AddChild( btnPhone )
	
	//Create a Get Contacts button.
	btnContact = app.CreateButton( "Get D Contacts", 0.4, 0.1 )
	btnContact.SetOnTouch( btnContact_OnTouch )
	lay.AddChild( btnContact )

	//Add layout to app.	
	app.AddLayout( lay )
}

//Called when user touches our Get Phone Number button.
function btnPhone_OnTouch()
{
    var action = "android.intent.action.PICK";
    var uri = "content://contacts";
    var type =  "vnd.android.cursor.dir/phone_v2";
    app.SendIntent( null, null, action, null, uri, type, null, "result", OnResult )
}

//Called with intent result.
function OnResult( resultCode, result )
{
    //Query for contact info.
    var rows = app.QueryContent( result.data, "data1, display_name" )
    app.ShowPopup( rows[0].display_name + " " + rows[0].data1 )
}

//Called when user touches our Get D Contact button.
function btnContact_OnTouch()
{
    //Query for all contact info where name begins with D.
    var uri = "content://com.android.contacts/data";
    var columns = "display_name, data1";
    var select = "display_name LIKE ?";
    var params = "D%";
    var rows = app.QueryContent( uri, columns, select, params )
    
    //Show result (stringify as useful way of converting JS objects to text)
    alert( JSON.stringify(rows) )
}

